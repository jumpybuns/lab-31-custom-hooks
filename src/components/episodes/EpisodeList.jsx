import React from 'react';
import { Link } from 'react-router-dom';
import Episode from './Episode';
import PropTypes from 'prop-types';

const EpisodeList = ({ episodes }) => {
  const episodeElements = episodes.map((episode) => (
    <li key={episode.id}>
      <Link to={`/episodes/${episode.id}`}>
        <Episode name={episode.name} url={episode.url} />
      </Link>
    </li>
  ));
  return (
    <div>
      <ul>{episodeElements}</ul>
    </div>
  );
};

export default EpisodeList;

EpisodeList.propTypes = {
  episodes: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired,
    })
  ).isRequired,
};
