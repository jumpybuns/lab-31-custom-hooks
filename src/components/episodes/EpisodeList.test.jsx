import React from 'react';
import { render, cleanup } from '@testing-library/react';
import EpisodeList from './EpisodeList';
import { MemoryRouter } from 'react-router-dom';

describe('EpisodeList component', () => {
  afterEach(() => cleanup());
  it('renders EpisodeList', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <EpisodeList
          episodes={[
            {
              id: 1,
              name: 'Pilot',
              url: 'https://rickandmortyapi.com/api/episode/1',
            },
          ]}
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
