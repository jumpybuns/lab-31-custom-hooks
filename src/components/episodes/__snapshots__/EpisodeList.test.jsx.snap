// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeList component renders EpisodeList 1`] = `
<DocumentFragment>
  <div>
    <ul
      data-testid="episodes"
    >
      <li>
        <a
          href="/episodes/1"
        >
          <figure>
            <h1>
              Pilot
            </h1>
            <h2>
              https://rickandmortyapi.com/api/episode/1
            </h2>
          </figure>
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
